samples/usb/main.c

```diff
@@ -0,0 +1,37 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <windows.h>
+#include <usbh_lib.h>
+
+/*
+ * This demo will initialise the usb stack, detect events on the usb ports
+ * and print the device PID and VID.
+*/
+void device_connection_callback(UDEV_T *udev, int status)
+{
+    debugPrint("Device connected on port %u (PID: %04x VID: %04x)\n", udev->port_num,
+                                                                      udev->descriptor.idProduct,
+                                                                      udev->descriptor.idVendor);
+}
+
+void device_disconnect_callback(UDEV_T *udev, int status)
+{
+     debugPrint("Device disconnected on port %u (PID: %04x VID: %04x)\n", udev->port_num,
+                                                                          udev->descriptor.idProduct,
+                                                                          udev->descriptor.idVendor);
+}
+
+int main(void)
+{
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+
+    usbh_core_init();
+    usbh_install_conn_callback(&device_connection_callback, &device_disconnect_callback);
+    debugPrint("Plug and unplug USB devices to test\n");

```

Comment:
This should be a bit more explicit.
People might assume that this will test their gamepad like that XDK sample that everybody loves (with the 3D gamepad).
This, however, merely seems to show plug and unplug events?

Reply:
updated the debug print here to make it clearer what it is doing